packages/nodes/emoji/src/handlers/emojiOnKeyDownHandler.ts

```diff
@@ -0,0 +1,53 @@
+import { KeyboardEvent } from 'react';
+import {
+  HandlerReturnType,
+  moveSelection,
+  PlateEditor,
+  Value,
+} from '@udecode/plate-core';
+import isHotkey from 'is-hotkey';
+import { Range } from 'slate';
+
+/**
+ * TODO: move to core
+ */
+type KeyboardEventHandler = (event: KeyboardEvent) => HandlerReturnType;
+
+interface MoveSelectionByOffsetOptions<V extends Value = Value> {
+  query?: (editor: PlateEditor<V>) => boolean;
+}
+
+// TODO: move to core
+const moveSelectionByOffset: <V extends Value>(
+  editor: PlateEditor<V>,
+  options?: MoveSelectionByOffsetOptions<V>
+) => KeyboardEventHandler = (editor, { query = () => true } = {}) => (
+  event
+) => {
+  const { selection } = editor;
+
+  if (!selection || Range.isExpanded(selection) || !query(editor)) {
+    return false;
+  }
+
+  if (isHotkey('left', event)) {
+    event.preventDefault();
+    moveSelection(editor, { unit: 'offset', reverse: true });
+    return true;
+  }
+
+  if (isHotkey('right', event)) {
+    event.preventDefault();
+    moveSelection(editor, { unit: 'offset' });
+    return true;
+  }
+};
+// ------>
+
+export const emojiOnKeyDownHandler: <V extends Value>(
```

Comment:
Can you rename it to `getOnKeyDownEmoji`, `mentionOnKeyDownHandler` didn't follow the naming pattern we have so it will change at some point too

Reply:
Renamed